packages/loopring_v3/circuit/Gadgets/MatchingGadgets.h

```diff
@@ -493,8 +493,8 @@ class OrderMatchingGadget : public GadgetT
 //     return Math.floor((ratio * BASE_BIPS) / invFeeBips);
 // }
 
-// Result is guaranteed to fit inside NUM_BITS_AMOUNT bits.
-// Max ratio between weights guaranteed to be supported is 2**(96-14)/(10**18) = 4835703.278458517
+// Result is guaranteed to fit inside NUM_BITS_AMOUNT*2 bits.
+// Max ratio between weights guaranteed to be supported is 2**(96*2-14)/(10**18)
```

Comment:
what if get rid of this check?

Reply:
The max number of bits of `ratio` is NUM_BITS_AMOUNT + NUM_BITS_AMOUNT + 60 (log2(baseFixed)) = 252`, which just fits inside a single field element. We still have to do `ratio * BASE_BIPS` which can increase the required number of bits by `14 = log2(10000)`, which could overflow because the results needs `252 + 14 = 266` bits. The highest we can make this is `NUM_BITS_FIELD_CAPACITY - 14`. But NUM_BITS_AMOUNT*2 seems more than large enough, no? If the weights are equal the absolute max ratio is just (NUM_BITS_AMOUNT + 60) bits. And even in this absolute worst case the weights would have to differ by a factor of 2^18.